### ISSUE TITLE
ERROR: InvalidStaticInvocation - RouteServiceProvider - Method Router::__construct is not static, but is called statically

### ISSUE THREAD

**Issue by caugner** [psalm/psalm-plugin-laravel#22]
I'm getting the following unexpected error on the `RouteServiceProvider` that comes with Laravel and hasn't been modified by me:

```
ERROR: InvalidStaticInvocation - app\Providers\RouteServiceProvider.php:12:36 - Method Illuminate\Routing\Router::__construct is not static, but is called statically
class RouteServiceProvider extends ServiceProvider
```

Unfortunately I cannot suppress it using `@psalm-suppress InvalidStaticInvocation` on the [`RouteServiceProvider`](https://github.com/laravel/laravel/blob/c4a7a7bc9cdccd00d6fdda8846d1660a0608d7ef/app/Providers/RouteServiceProvider.php) class either.

<details>
<summary>RouteServiceProvider.php</summary>

```php
<?php

namespace App\Providers;

use Illuminate\Support\Facades\Route;
use Illuminate\Foundation\Support\Providers\RouteServiceProvider as ServiceProvider;

class RouteServiceProvider extends ServiceProvider
{
    /**
     * This namespace is applied to your controller routes.
     *
     * In addition, it is set as the URL generator's root namespace.
     *
     * @var string
     */
    protected $namespace = 'App\Http\Controllers';

    /**
     * Define your route model bindings, pattern filters, etc.
     *
     * @return void
     */
    public function boot()
    {
        //

        parent::boot();
    }

    /**
     * Define the routes for the application.
     *
     * @return void
     */
    public function map()
    {
        $this->mapApiRoutes();

        $this->mapWebRoutes();

        //
    }

    /**
     * Define the "web" routes for the application.
     *
     * These routes all receive session state, CSRF protection, etc.
     *
     * @return void
     */
    protected function mapWebRoutes()
    {
        Route::middleware('web')
             ->namespace($this->namespace)
             ->group(base_path('routes/web.php'));
    }

    /**
     * Define the "api" routes for the application.
     *
     * These routes are typically stateless.
     *
     * @return void
     */
    protected function mapApiRoutes()
    {
        Route::prefix('api')
             ->middleware('api')
             ->namespace($this->namespace)
             ->group(base_path('routes/api.php'));
    }
}
```

</details>

**Comment by shufo** [comment]
@caugner 
same here.
To workaround this, I did set baseline.

```bash
$ ./vendor/bin/psalm --set-baseline=psalm-baseline.xml
```

**Comment by caugner** [comment]
Thanks, looks like this is actually not a plugin problem, also occurs in a fresh Laravel site without this plugin.
